Subject: [PATCH] Fix __ucmpdi2 in v4l2_norm_to_name()                                                                                                                
From: Stelian Pop <stelian@popies.net>                                                                                                                               

This patch replaces a switch statement using 64 bit values with the
if/else equivalent in order to prevent a call __ucmpdi2 generated by
some versions of gcc (verified with gcc-4.1.2 20060928):

        drivers/built-in.o: In function `v4l2_norm_to_name':
        (.text+0x71100): undefined reference to `__ucmpdi2'

Signed-off-by: Olaf Hering <olh@suse.de>

---
 drivers/media/video/v4l2-common.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/drivers/media/video/v4l2-common.c
===================================================================
--- a/drivers/media/video/v4l2-common.c
+++ b/drivers/media/video/v4l2-common.c
@@ -91,7 +91,7 @@ char *v4l2_norm_to_name(v4l2_std_id id)
 {
 	char *name;
 
-	switch (id) {
+	switch ((unsigned int)id) {
 	case V4L2_STD_PAL:
 		name="PAL";		break;
 	case V4L2_STD_PAL_BG:
